eng/scripts/verify-npm-tags.ps1

```diff
@@ -0,0 +1,59 @@
+param (
+  [Parameter(mandatory = $true)]
+  $artifactName,
+  [Parameter(mandatory = $true)]
+  $workingDirectory
+)
+
+# TODO: delete testing comments below
+#'@azure/eventhubs-checkpointstore-table' alpha, no beta, no GA
+#'@azure/openai' alpha, beta, no GA
+#'@azure/video-indexer-widgets' no alpha, no beta, GA
+#'@azure/template' alpha, beta, GA
+#$pkgProps.PackageId
+
+. (Join-Path $PSScriptRoot "../common/scripts/common.ps1")
+
+$packageProperties = Get-PkgProperties -PackageName $artifactName
+$packageName =  $packageProperties.Name
+Write-Host "Verify npm tag versions for package $packageName"
+$packageVersions = npm view $packageName versions --json | ConvertFrom-Json
+$validDev = $packageVersions | ? { $_ -match "alpha" } | Select-Object -Last 1
```

Comment:
We should be validating the tags we are setting. It seems you are trying to validate the current state of the package on npm as opposed to validating the tags we think we just set.

Reply:
is there a reason why we'd want to validate the tags we're setting instead of the current state of the package?

we'll still need to correct it anyway. For example, we're setting v10.0.0-beta with next tag, but the pipeline set it to latest.
we'll still need to revert the change by getting the "correct latest" of v10.0.0

Reply:
The main reason for this validation is to ensure that the ESRP task is doing what we ask them to do. What happens if the tags are in an invalid state before we even do this run? This release could be blocked because of something unrelated to publishing.

Reply:
Also we need to keep in mind that verifying all the tags in this way is JS repos specific and doesn't accomplish the goal of verifying the tags that we expected ESRP to set when we publish another package outside of the JS sdk repo context.

Reply:
what do we want to do then in the case of an incorrectly marked tag?

e.g.
1.0.0 latest
1.0.0-alpha dev
1.0.0-beta next

and esrp accidentally marked
1.0.1-beta latest
1.0.0-alpha dev
1.0.0-beta next

we'll still need to change 1.0.0 back to latest and 1.0.2-beta to next right?
npm also dont keep a record of the "past versions with X tag" so we can't trace back from history unfortunately

Reply:
and i thought our consensus was that the verify step won't halt on tag mismatch? just correct it and we can get metrics of when it failed by filtering the logs